src/pybind/mgr/zabbix/module.py

```diff
@@ -0,0 +1,204 @@
+
+        wr_ops = 0
+        rd_ops = 0
+        wr_bytes = 0
+        rd_bytes = 0
+
+        for pool in df['pools']:
+            wr_ops += pool['stats']['wr']
+            rd_ops += pool['stats']['rd']
+            wr_bytes += pool['stats']['wr_bytes']
+            rd_bytes += pool['stats']['rd_bytes']
+
+        data['wr_ops'] = wr_ops
+        data['rd_ops'] = rd_ops
+        data['wr_bytes'] = wr_bytes
+        data['rd_bytes'] = rd_bytes
+
+        osd_map = self.get('osd_map')
+        data['num_osd'] = len(osd_map['osds'])
+        data['osd_nearfull_ratio'] = osd_map['nearfull_ratio']
+        data['osd_full_ratio'] = osd_map['full_ratio']
+        data['osd_backfillfull_ratio'] = osd_map['backfillfull_ratio']
+
+        data['num_pg_temp'] = len(osd_map['pg_temp'])
+
+        num_up = 0
+        num_in = 0
+        for osd in osd_map['osds']:
+            if osd['up'] == 1:
+                num_up += 1
+
+            if osd['in'] == 1:
+                num_in += 1
+
+        data['num_osd_up'] = num_up
+        data['num_osd_in'] = num_in
+
+        osd_fill = list()
+
+        osd_stats = self.get('osd_stats')
+        for osd in osd_stats['osd_stats']:
+            osd_fill.append((osd['kb_used'] / osd['kb']) * 100)
+
+        if len(osd_fill) > 0:
+            data['osd_max_fill'] = max(osd_fill)
+            data['osd_min_fill'] = min(osd_fill)
+            data['osd_avg_fill'] = sum(osd_fill) / float(len(osd_fill))
+
+        pg_summary = self.get('pg_summary')
+        num_pg = 0
+        for state, num in pg_summary['all'].items():
+            num_pg += num
+
+        data['num_pg'] = num_pg
+
+        return data
+
+    def shutdown(self):
+        self.log.info('Stopping zabbix')
+        self.run = False
+        self.event.set()
+
+    def serve(self):
+        self.log.debug('Zabbix module starting up')
+        self.run = True
+
+        conf = self.get_module_config()
+
+        for key, value in conf.items():
+            self.log.debug('%s: %s', key, value)
+
+        zabbix = ZabbixSender(conf['zabbix_sender'], conf['zabbix_host'],
+                              conf['zabbix_port'], self.log)
+
+        while self.run:
+            self.log.debug('Waking up for new iteration')
+
+            data = self.get_data()
+
+            self.log.debug('Sending data to Zabbix server %s',
+                           conf['zabbix_host'])
+            self.log.debug(data)
+            try:
+                zabbix.send(conf['identifier'], data)
+            except Exception as exc:
+                self.log.error('Exception when sending: %s', exc)
+
+            self.log.debug('Sleeping for {0} seconds'.format(conf['interval']))
+            self.event.wait(conf['interval'])
+
+    def self_test(self):
+        data = self.get_data()
+
+        if data['overall_status'] not in self.ceph_health_mapping:
+            raise RuntimeError('No valid overall_status found in data')
+
+        return True
```

Comment:
@wido seems you are raising an exception if `self_test()` does not pass, that why i suggest just return nothing if everything looks good.